### DISCUSSION TITLE
ListBox Multiple

### DISCUSSION THREAD

**Discussion by PlayerModu** [microsoft/fluentui-blazor#756] [Q&A] [answered]
The documentation on the ComboBox (https://www.fluentui-blazor.net/Combobox) documentation it states that...

"If true, the user can select multiple elements.⚠️ Only available for the FluentSelect and FluentListbox components."

However, when looking at the ListBox (https://www.fluentui-blazor.net/Listbox) documentation, it states:

![image](https://github.com/microsoft/fluentui-blazor/assets/65035377/3de55f20-ebd9-4af4-bd9a-d93a06880e6e)

Is there a timeline for when this will be delivered please? 

Also are there plans to make the ComboBox support Multiple once this has been implemented?

**Comment by vnbaaij** [comment]
No, there is no time line for when this will be fixed/done.

Combobox will never be multiple as that is not what it is supposed to to. You either select 1 item from the list OR you type a value yourself. That's it.
You might want to look at the autocomplete component (available in v3. 1)

**Reply by PlayerModu** [reply]
Hi @vnbaaij,

Thank you for your response, I understand there is no timeline for the delivery of this feature.

I can't see AutoComplete in the current documentation, you noted Version 3.1 above I'm guessing this is a future release? Apologies if I'm simply just not finding the component, I fairly new to this particular Blazor component library so just finding my way around it.

As a general question, why wouldn't ComboBox look to support Multiple in the future? I ask because I'm wondering about this example scenario and how to solve it with this library: I want to search a full list of countries for those countries whose name contains the letter A, and then be able to select just those which are filtered?

**Reply by vnbaaij** [reply, marked as answer]
No worries, we are here to help!

The next version of the library is (almost always) available at https://preview.fuentui-blazor.net. You can find the docs and examples for the autocomplete there (https://preview.fluentui-blazor.net/Autocomplete)

What you are describing is possible with the Autocomplete component. Like I tried to explain, that is not something that a Combobox can do. You always only select 1 value (either from the list or self entered)

**Reply by PlayerModu** [reply]
Hi @vnbaaij,

Thank you for sharing those links, I can see that the AutoComplete component you have linked fits exactly the scenario I was describing! I am glad to see this in progress :) and I understand why ComboBox will not support this.

Is there any roadmap currently on when this component is planned to be released out of the preview build?

And if I may ask another question on this AutoComplete component. After selecting an item, the list automatically closes and then I either need to re-type into the search or click and scroll through the list again to select another item. Is there any possibility of keeping the list open after selecting so that I can simply click/search once and then click through each desired item to be selected? Perhaps this could be an option on the component?

**Reply by vnbaaij** [reply]
We intend to release v3.1 this week (maybe already tomorrow). I'm looping in @dvoituron (author of the component) on answering the 'keeping the list open' question,

**Reply by PlayerModu** [reply]
Ah excellent news :) I shall keep an eye out for the pending release. Thanks again for your quick and helpful replies.

Happy to discuss the "keeping the list open" question further if/when needed.

**Reply by dvoituron** [reply]
> Perhaps this could be an option on the component?

Not yet, but it's a good idea. I'll add it to the backlog :-)

**Reply by PlayerModu** [reply]
@dvoituron okay thank you ☺️

**Reply by markarnolditpro** [reply]
@dvoituron  +1 for ListBox multi select
